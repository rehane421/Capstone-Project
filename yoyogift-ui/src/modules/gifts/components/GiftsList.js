// import React from "react";
// import PropTypes from "prop-types";
// import { withStyles } from "@material-ui/core/styles";
// import Table from "@material-ui/core/Table";
// import TableBody from "@material-ui/core/TableBody";
// import TableCell from "@material-ui/core/TableCell";
// import TableFooter from "@material-ui/core/TableFooter";
// import TablePagination from "@material-ui/core/TablePagination";
// import TableRow from "@material-ui/core/TableRow";
// import Paper from "@material-ui/core/Paper";
// import Grid from "@material-ui/core/Grid";
// import { Button } from "@material-ui/core";
// import { List } from "react-virtualized";
// import GiftCard from "../../common/components/GiftCard";
// import { TablePaginationActionsWrapped } from "../../common/components/TablePaginationActionsWrapped";

// const styles = theme => ({
//   root: {
//     width: "90%",
//     margin: "2%",
//     padding: "1%"
//   },
//   table: {
//     minWidth: 100
//   },
//   tableWrapper: {
//     overflow: "hidden"
//   }
// });

// export class GiftsList extends React.Component {
//   state = {
//     giftCardsFiltered: this.props.giftCardsFiltered,
//     page: 0,
//     rowsPerPage: 10
//   };

//   rowRenderer = ({
//     key, // Unique key within array of rows
//     index, // Index of row within collection
//     isScrolling, // The List is currently being scrolled
//     isVisible, // This row is visible within the List (eg it is not an overscanned row)
//     style // Style object to be applied to row (to position it)
//   }) => {
//     let { giftCardsFiltered, userDetails } = this.props;
//     return (
//       <div key={key} style={style}>
//         <Grid container spacing={16}>
//           <p>{index}</p>
//           <Grid item xs={12} sm={6} md={3} lg={3}>
//             <GiftCard
//               giftCard={giftCardsFiltered[index]}
//               userEmail={userDetails.email}
//             />
//             {userDetails.email === "lathak95@gmail.com" ||
//             this.props.userDetails.email === "yoyogiftg2@gmail.com" ? (
//               <Button
//                 variant="contained"
//                 color="primary"
//                 onClick={() =>
//                   this.props.handleUpdateClick(giftCardsFiltered[index].id)
//                 }
//               >
//                 UPDATE
//               </Button>
//             ) : null}
//           </Grid>
//         </Grid>
//       </div>
//     );
//   };

//   // handleChangePage = (event, page) => {
//   //   this.setState({ page });
//   // };

//   // handleChangeRowsPerPage = event => {
//   //   this.setState({ page: 0, rowsPerPage: event.target.value });
//   // };

//   render() {
//     // let { classes, giftCardsFiltered, userDetails } = this.props;
//     // const { rowsPerPage, page } = this.state;
//     // const emptyRows =
//     //   rowsPerPage -
//     //   Math.min(rowsPerPage, giftCardsFiltered.length - page * rowsPerPage);
//     let { giftCardsFiltered } = this.props;
//     return (
//       // <Paper className={classes.root}>
//       //   <div className={classes.tableWrapper}>
//       //     <Table className={classes.table}>
//       //       <TableBody>
//       //         <Grid container spacing={16}>
//       //           {giftCardsFiltered
//       //             .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
//       //             .map(giftCard => {
//       //               return (
//       //                 <Grid item xs={12} sm={6} md={3} lg={3}>
//       //                   <GiftCard
//       //                     giftCard={giftCard}
//       //                     userEmail={userDetails.email}
//       //                   />
//       //                   {/* {(userDetails.email === "lathak95@gmail.com" || this.props.userDetails.email === "yoyogiftg2@gmail.com")? <Button variant="contained" color="primary" onClick={()=> this.props.handleUpdateClick(giftCard.id)}>UPDATE</Button> : null} */}
//       //                 </Grid>
//       //               );
//       //             })}
//       //         </Grid>
//       //         {emptyRows > 0 && (
//       //           <TableRow style={{ height: 48 * emptyRows }}>
//       //             <TableCell colSpan={6} />
//       //           </TableRow>
//       //         )}
//       //       </TableBody>
//       //       <TableFooter>
//       //         <TableRow>
//       //           <TablePagination
//       //             rowsPerPageOptions={[10, 20]}
//       //             colSpan={3}
//       //             count={giftCardsFiltered.length}
//       //             rowsPerPage={rowsPerPage}
//       //             page={page}
//       //             SelectProps={{
//       //               native: true
//       //             }}
//       //             onChangePage={this.handleChangePage}
//       //             onChangeRowsPerPage={this.handleChangeRowsPerPage}
//       //             ActionsComponent={TablePaginationActionsWrapped}
//       //             props
//       //           />
//       //         </TableRow>
//       //       </TableFooter>
//       //     </Table>
//       //   </div>
//       // </Paper>
//       <div alignItems="center">
//         <List
//           width={1000}
//           height={600}
//           rowHeight={315}
//           rowRenderer={this.rowRenderer}
//           rowCount={giftCardsFiltered.length}
//         />
//       </div>
//     );
//   }
// }

// GiftsList.propTypes = {
//   classes: PropTypes.object.isRequired
// };

// export default withStyles(styles)(GiftsList);

import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableFooter from "@material-ui/core/TableFooter";
import TablePagination from "@material-ui/core/TablePagination";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import { List } from "react-virtualized";
import { Button } from "@material-ui/core";
import GiftCard from "../../common/components/GiftCard";
import { TablePaginationActionsWrapped } from "../../common/components/TablePaginationActionsWrapped";

const styles = theme => ({
  root: {
    width: "90%",
    margin: "2%",
    padding: "1%"
  },
  table: {
    minWidth: 100
  },
  tableWrapper: {
    overflow: "hidden"
  }
});

class GiftsList extends React.Component {
  state = {
    giftCardsFiltered: this.props.giftCardsFiltered,
    searchedCard: this.props.searchedCard,
    page: 0,
    rowsPerPage: 10
  };
  rowRenderer = ({
    key, // Unique key within array of rows
    index, // Index of row within collection
    isScrolling, // The List is currently being scrolled
    isVisible, // This row is visible within the List (eg it is not an overscanned row)
    style // Style object to be applied to row (to position it)
  }) => {
    let { giftCardsFiltered, searchedCard, userDetails } = this.props;
    return (
      <div key={key} style={style}>
        {searchedCard
          // .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
          .map(giftCard => {
            return (
              <Grid item xs={12} sm={6} md={3} lg={3}>
                <GiftCard giftCard={giftCard} userEmail={userDetails.email} />
                {/* {(userDetails.email === "lathak95@gmail.com" || this.props.userDetails.email === "yoyogiftg2@gmail.com")? <Button variant="contained" color="primary" onClick={()=> this.props.handleUpdateClick(giftCard.id)}>UPDATE</Button> : null} */}
              </Grid>
            );
          })}
      </div>
    );
  };

  handleChangePage = (event, page) => {
    this.setState({ page });
  };

  handleChangeRowsPerPage = event => {
    this.setState({ page: 0, rowsPerPage: event.target.value });
  };

  render() {
    let { classes, giftCardsFiltered, userDetails, searchedCard } = this.props;
    // const { rowsPerPage, page } = this.state;
    // const emptyRows =
    //   rowsPerPage -
    //   Math.min(rowsPerPage, searchedCard.length - page * rowsPerPage);
    return (
      <Paper className={classes.root}>
        <div className={classes.tableWrapper}>
          <Table className={classes.table}>
            <TableBody>
              <Grid container spacing={16}>
                <List
                  width={1000}
                  height={600}
                  rowHeight={315}
                  rowRenderer={this.rowRenderer}
                  rowCount={giftCardsFiltered.length}
                />
              </Grid>
              {/* {emptyRows > 0 && (
                <TableRow style={{ height: 48 * emptyRows }}>
                  <TableCell colSpan={6} />
                </TableRow>
              )} */}
            </TableBody>
            {/* <TableFooter>
              <TableRow>
                <TablePagination
                  rowsPerPageOptions={[10, 20]}
                  colSpan={3}
                  count={giftCardsFiltered.length}
                  rowsPerPage={rowsPerPage}
                  page={page}
                  SelectProps={{
                    native: true
                  }}
                  onChangePage={this.handleChangePage}
                  onChangeRowsPerPage={this.handleChangeRowsPerPage}
                  ActionsComponent={TablePaginationActionsWrapped}
                  props
                />
              </TableRow>
            </TableFooter> */}
          </Table>
        </div>
      </Paper>
    );
  }
}

GiftsList.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(GiftsList);
