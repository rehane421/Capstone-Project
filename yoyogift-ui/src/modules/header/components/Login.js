import React, { useState } from "react";
import { login } from "../state/actions/index";
import { GoogleLogin } from "react-google-login";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
// import TextField from "@material-ui/core/TextField";
import TextFieldAtom from "../../atoms/TextField/TextFieldAtom";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import PermIdentityTwoToneIcon from "@material-ui/icons/PermIdentityTwoTone";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { google } from "../../../config/constants";
import { connect } from "react-redux";

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(5),
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: "#12AE96"
  },
  form: {
    // width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1)
  },
  submit: {
    backgroundColor: "#12AE96",
    textAlign: "center",
    width: "100%",
    margin: theme.spacing(3, 0, 2)
  },
  mainContainer: {
    width: 300
  }
}));

export function Login({ login }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const classes = useStyles();

  const responseGoogle = response => {
    login(response);
  };
  const loginSubmit = e => {
    e.preventDefault();
    let data = { email: email, password: password };
    login(data);
  };
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justify="center"
      style={{ minHeight: "100vh" }}
    >
      <Grid item xs={10}>
        <div component="main" className={classes.mainContainer}>
          <CssBaseline />
          <div className={classes.paper}>
            <Avatar className={classes.avatar}>
              <PermIdentityTwoToneIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
              Sign in
            </Typography>
            <form className={classes.form} noValidate>
              <TextFieldAtom
                variant="outlined"
                margin="normal"
                required
                fullWidth
                id="email"
                label="Email Address"
                name="email"
                autoComplete="email"
                autoFocus
                onChange={e => {
                  setEmail(e.target.value);
                }}
                setEmail={setEmail}
              />
              <TextFieldAtom
                variant="outlined"
                margin="normal"
                required
                fullWidth
                name="password"
                label="Password"
                type="password"
                id="password"
                autoComplete="current-password"
                onChange={e => {
                  setPassword(e.target.value);
                }}
                setPassword={setPassword}
              />
              {/* <FormControlLabel
                control={<Checkbox value="remember" color="primary" />}
                label="Remember me"
              /> */}
              <Button
                id="xyz"
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                className={classes.submit}
                onClick={e => loginSubmit(e)}
              >
                Sign In
              </Button>
              <GoogleLogin
                className={classes.submit}
                clientId={google.loginClientId}
                buttonText="Continue with Google"
                onSuccess={responseGoogle}
                onFailure={responseGoogle}
                cookiePolicy={"single_host_origin"}
              />

              {/* <Grid container>
                <Grid item sm={12}>
                  <Link href="#" variant="body2">
                    Forgot password?
                  </Link>
                </Grid>
                <Grid item>
                  <Link href="#" variant="body2">
                    {"Don't have an account? Sign Up"}
                  </Link>
                </Grid>
              </Grid> */}
            </form>
          </div>
          <Box mt={8}></Box>
        </div>
      </Grid>
    </Grid>
  );
}

export default connect(null, { login })(Login);
