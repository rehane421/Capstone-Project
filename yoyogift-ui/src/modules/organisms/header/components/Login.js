import React from "react";
import { login } from "../state/actions/index";
import { GoogleLogin } from "react-google-login";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import PermIdentityTwoToneIcon from "@material-ui/icons/PermIdentityTwoTone";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { google } from "../../../../config/constants";
import { connect } from "react-redux";
import { Formik } from "formik";
import * as Yup from "yup";

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(5),
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: "#12AE96"
  },
  form: {
    marginTop: theme.spacing(1)
  },
  submit: {
    backgroundColor: "#12AE96",
    textAlign: "center",
    width: "100%",
    margin: theme.spacing(3, 0, 2)
  },
  mainContainer: {
    width: 300
  }
}));

export function Login({ login }) {
  // const [email, setEmail] = useState("");
  // const [password, setPassword] = useState("");
  const classes = useStyles();

  const responseGoogle = response => {
    login(response);
  };
  const loginSubmit = values => {
    let data = { email: values.email, password: values.password };
    login(data);
  };
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justify="center"
      style={{ minHeight: "100vh" }}
    >
      <Grid item xs={10}>
        <div component="main" className={classes.mainContainer}>
          <CssBaseline />
          <div className={classes.paper}>
            <Avatar className={classes.avatar}>
              <PermIdentityTwoToneIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
              Sign in
            </Typography>
            <Formik
              initialValues={{
                email: "",
                password: ""
              }}
              onSubmit={values => loginSubmit(values)}
              validationSchema={Yup.object().shape({
                email: Yup.string()
                  .email("Invalid email")
                  .required("Required"),
                password: Yup.string().required("required")
              })}
            >
              {props => {
                const {
                  values,
                  touched,
                  errors,
                  handleChange,
                  handleBlur,
                  handleSubmit
                } = props;
                return (
                  <form onSubmit={handleSubmit}>
                    <TextField
                      variant="outlined"
                      margin="normal"
                      fullWidth
                      id="email"
                      label="Email Address"
                      name="email"
                      error={errors.email && touched.email}
                      onChange={handleChange}
                      onBlur={handleBlur}
                      helperText={errors.email && touched.email && errors.email}
                      value={values.email}
                    />
                    <TextField
                      variant="outlined"
                      margin="normal"
                      fullWidth
                      name="password"
                      label="Password"
                      type="password"
                      id="password"
                      autoComplete="current-password"
                      error={errors.password && touched.password}
                      onChange={handleChange}
                      onBlur={handleBlur}
                      helperText={
                        errors.password && touched.password && errors.password
                      }
                      value={values.password}
                    />
                    <Button
                      id="xyz"
                      type="submit"
                      fullWidth
                      variant="contained"
                      color="primary"
                      className={classes.submit}
                    >
                      Sign In
                    </Button>
                    <GoogleLogin
                      className={classes.submit}
                      clientId={google.loginClientId}
                      buttonText="Continue with Google"
                      onSuccess={responseGoogle}
                      onFailure={responseGoogle}
                      cookiePolicy={"single_host_origin"}
                    />
                  </form>
                );
              }}
            </Formik>
          </div>
          <Box mt={8}></Box>
        </div>
      </Grid>
    </Grid>
  );
}

export default connect(null, { login })(Login);
