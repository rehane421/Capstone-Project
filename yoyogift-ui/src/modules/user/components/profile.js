import React from "react";
import Grid from "@material-ui/core/Grid";
import Styles from "../../../assets/css/Profile.module.css";
import Paper from "@material-ui/core/Paper";

export const Profile = props => {
  let {
    email,
    first_name,
    last_name,
    socialProfileLink,
    image,
    balance_points
  } = props.detailsObject;
  return (
    <React.Fragment>
      <br />
      <br />
      {/* <div className="row">
        <div className="col-md-12">
          {" "}
          <label className="">
            <h1>Your Profile</h1>{" "}
          </label>
        </div>
      </div>
      <hr />
      <div className="row">
        <div className="col-md-12">
          {" "}
          <label className="">
            {" "}
            <img
              className="img-fluid"
              alt="profile"
              src={image}
              height="150"
              width="150"
            />
          </label>
        </div>
      </div>
      <hr />
      <div className="row">
        <div className="col-md-4">
          {" "}
          <label className="">Name</label>
        </div>
        <div className="col-md-8">
          {" "}
          <label className="">
            {" "}
            {firstName} {lastName}
          </label>
        </div>
      </div>
      <hr />
      <div className="row">
        <div className="col-md-4">
          {" "}
          <label className="">Email </label>
        </div>
        <div className="col-md-8">
          {" "}
          <label className="">{email} </label>
        </div>
      </div>
      <hr />
      <div className="row">
        <div className="col-md-4">
          {" "}
          <label className="">Gender </label>
        </div>
        <div className="col-md-8">
          {" "}
          <label className="">gender </label>
        </div>
      </div>
      <hr />
      <div className="row">
        <div className="col-md-4">
          {" "}
          <label className="">Social Link </label>
        </div>
        <div className="col-md-8">
          {" "}
          <label className="">
            {" "}
            <a
              target="_blank"
              rel="noopener noreferrer"
              href={socialProfileLink}
            >
              {socialProfileLink}
            </a>
          </label>
        </div>
      </div>
      <hr /> */}
      <div className={Styles.profileHeader}>
        <img className={Styles.profileImg} alt="profile" src={image} />
      </div>
      <div className={Styles.profileContent}>
        <Grid container spacing={0}>
          <Grid item xs={4}>
            <label className={Styles.label}>
              Name: {first_name} {last_name}
            </label>
            <label className={Styles.label}>User Id: {email}</label>
            {socialProfileLink ? (
              <label className={Styles.label}>Social Profile Contact</label>
            ) : null}
            <label className={Styles.label}>
              Balance Points: {balance_points}
            </label>
          </Grid>
          <Grid item xs={8}>
            {socialProfileLink ? (
              <Paper className={Styles.profileValues}>
                {socialProfileLink}
              </Paper>
            ) : null}
          </Grid>
        </Grid>
      </div>
    </React.Fragment>
  );
};

export default Profile;
