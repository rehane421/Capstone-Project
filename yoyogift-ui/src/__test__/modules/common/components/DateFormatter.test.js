import React from "react";
import { shallow } from "../../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { DateFormatter } from "../../../../modules/common/components/DateFormatter";

describe("Date Formatter", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      date: new Date("2/14/2020")
    };
  });

  it("Should render Footer component", () => {
    wrapper = shallow(<DateFormatter {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("Should render Footer component with month length < 2", () => {
    let wrapper = shallow(<DateFormatter {...props} />);
    // expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper);
  });
});
