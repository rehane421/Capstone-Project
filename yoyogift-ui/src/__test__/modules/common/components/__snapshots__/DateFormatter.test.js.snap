// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Formatter Should render Footer component 1`] = `"NaN-NaN-NaN"`;
